### ISSUE TITLE
Responsive design considerations for different viewport sizes

### ISSUE THREAD

**Issue by jscholes** [benetech/ImageShare#50]
This is a higher-level thread for discussion about responsive design considerations to make Imageshare usable and accessible on both small and large screens.

Initial notes from @clapierre:

> Yes that would be important I think as well.  So on a smaller screen we maybe only see 2 x 2 collections etc. and a vertical scroller to show more.

**Comment by sinabahram** [comment]
I’d like to lock this in early, because this can be a huge sink of effort and we are on a fixed schedule for this project. We should classify things like scrolling on mobile for a nice-to-have, but basic responsive should be considered in scope e.g. it reflows, but a dedicated scroller or vertical navigation is in the extra column, for sure.

 

Let me know if that makes sense.

**Comment by clapierre** [comment]
I was thinking that there may be a large number of curated collections and that we would only see a certain # of them on the main page but you could scroll if you wanted to see more.  If we just reflow that then if we get a couple hundred curated collections that will make that initial page pretty unwieldily.   There is no easy drop-in for a scrollable view area either vertical or some horizontal accessible carousel that we can use for this?

**Comment by sinabahram** [comment]
There’s a far more fundamental concern there. If you have hundreds of collections, we shouldn’t display anything past 10 or so of them on the homepage, on any platform, so mobile has nothing to do with it.

 

So, I recommend having 10 or so on the homepage, then “See All Collections”, and then on that subsequent page, we can list them in 10, 20, or 50 at a time, with a very straight forward pagination control.

 

Thoughts?

**Comment by clapierre** [comment]
Since we already have that option if a resource is found in multiple collections you can select that and see a list of all of collections this resource belongs to, so this fits in with that requirement.

I guess if we have the ability to select the "featured" collections and show those only on the main landing page, that could work, and I think is in line with what we were originally thinking when we added collections.

As long as search also now includes the collection titles as well so someone searching for "Biology" will find the various Biology collections, as well as a resource that was tagged with "Biology" or had Biology in the title etc.

**Comment by sinabahram** [comment]
Correct, search includes collections titles, but we need to agree how to differentiate collections from resources in search results, which should be easy.

**Comment by sinabahram** [comment]
I think this thread has served its purpose for now, but please note, I am not claiming responsive design issues are fully done, only that we should open up a new issue for further discussion when the time is write to discuss different responsive design considerations.
